File: src/CHTL/Node/NamespaceNode.java
package CHTL.Node;

import java.util.List;
import java.util.ArrayList;

// Represents a [Namespace] block
public class NamespaceNode implements BaseNode {
    private final String name;
    private final List<BaseNode> children;

    public NamespaceNode(String name) {
        this.name = name;
        this.children = new ArrayList<>();
    }

    public void addChild(BaseNode node) {
        this.children.add(node);
    }

    public String getName() {
        return name;
    }

    public List<BaseNode> getChildren() {
        return children;
    }

    @Override
    public BaseNode clone() {
        NamespaceNode cloned = new NamespaceNode(name);
        for (BaseNode child : children) {
            cloned.addChild(child.clone());
        }
        return cloned;
    }
}

File: CHTL/CHTLNode/NamespaceNode.cpp
#include "NamespaceNode.h"
#include "Visitor.h"

namespace CHTL {

NamespaceNode::NamespaceNode(std::string name) : name(std::move(name)) {}

void NamespaceNode::addChild(std::unique_ptr<BaseNode> child) {
    children.push_back(std::move(child));
}

void NamespaceNode::accept(Visitor& visitor) {
    visitor.visit(*this);
}

std::unique_ptr<BaseNode> NamespaceNode::clone() const {
    auto new_node = std::make_unique<NamespaceNode>(name);
    for (const auto& child : children) {
        new_node->addChild(child->clone());
    }
    return new_node;
}

} // namespace CHTL
